modules/pubxaiAnalyticsAdapter.js

```diff
@@ -43,81 +47,48 @@ var pubxaiAnalyticsAdapter = Object.assign(adapter(
         }, 'bidwon');
       }
     }
-
     if (eventType === CONSTANTS.EVENTS.AUCTION_END) {
       send(events, 'auctionEnd');
     }
   }
 });
 
-function mapBidRequests(params) {
-  let arr = [];
-  if (typeof params.bids !== 'undefined' && params.bids.length) {
-    params.bids.forEach(function (bid) {
-      arr.push({
-        bidderCode: bid.bidder,
-        bidId: bid.bidId,
-        adUnitCode: bid.adUnitCode,
-        requestId: bid.bidderRequestId,
-        auctionId: bid.auctionId,
-        placementId: bid.params.placementId,
-        floorData: bid.floorData,
-        transactionId: bid.transactionId,
-        sizes: utils.parseSizesInput(bid.mediaTypes.banner.sizes).toString(),
-        renderStatus: 1,
-        requestTimestamp: params.auctionStart
-      });
-    });
-  }
-  return arr;
-}
-
 function mapBidResponse(bidResponse, status) {
-  if (status !== 'bidwon') {
-    let bid = events.bids.filter(o => o.bidId === bidResponse.bidId || o.bidId === bidResponse.requestId)[0];
-    Object.assign(bid, {
-      bidderCode: bidResponse.bidder,
-      bidId: status === 'timeout' ? bidResponse.bidId : bidResponse.requestId,
+  if (typeof bidResponse !== 'undefined' && bidResponse) {
```

Comment:
Not sure if there is a need to check if type is undefined AND also if the value is truthy.

I think just one of the other is enough.